digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/pointasset/ObstacleService.scala

```diff
@@ -41,9 +41,9 @@ class ObstacleService(val roadLinkService: RoadLinkService) extends PointAssetOp
     }
   }
 
-  override def update(id: Long, updatedAsset: IncomingObstacle, geometry: Seq[Point], municipality: Int, username: String, linkSource: LinkGeomSource): Long = {
+  override def update(id: Long, updatedAsset: IncomingObstacle, roadLink: RoadLink, username: String): Long = {
     withDynTransaction {
-      updateWithoutTransaction(id, updatedAsset, geometry, municipality, username, linkSource, None, None)
+      updateWithoutTransaction(id, updatedAsset, roadLink.geometry, roadLink.municipalityCode, username, roadLink.linkSource, None, None)
```

Comment:
instead of , ...roadLink.geometry, roadLink.municipalityCode, username, roadLink.linkSource ...we could only as a parameter roadLink.